### DISCUSSION TITLE
Any plan to support Match Whole Word?

### DISCUSSION THREAD

**Discussion by SunHuawei** [community/community#10608] [Other Feature Feedback, Questions, & Ideas] [answered]
Do you have a plan to support "Match Whole Word" which is the most popular feature?
I tried the "Exact match" feature as the home page described, but it doesn't work as expected. For example, when I [search by "sourcemap"](https://cs.github.com/?scopeName=All+repos&scope=&q=%22sourcemap%22), I expect the results won't include 
1. `sourceMap` (maybe this should be covered by some feature called Match Case)
2. `SourceMapGenerator`
3. `sourcemaps`

**Comment by colinwm** [comment, marked as answer]
A couple things:

1. We _rank_ whole word matches higher than partial matches. So hopefully the top results will be whole word matches. 
2. We don't support case sensitive matching yet (but we're working on it!)

If you want to strictly find whole word matches, you can also construct a regular expression. E.g. to find sourcemap, you could try something like `/(^|\W)sourcemap($|\W)/`

**Comment by RoyTinker** [comment]
This would make sense as a flag. It's a common feature in editors - I found myself trying several things to try to make it happen.

Perhaps internally it could function by translating each term `someTerm` to `/(^|\W)someTerm($|\W)/` before processing the query?
